src/binWrapper.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * Note:
+ * This is a modified fork of https://github.com/kevva/bin-wrapper
+ * The module had a CVE-2021-3795 vulnerability and decided to fork
+ * and "simplify" the implementation
+ */
+import {stat, chmod} from 'fs/promises'
+import {join, parse} from 'path'
+import url from 'url'
+import download from 'download'
+import osFilterObj  from 'os-filter-obj'
+
+export default class BinWrapper {
```

Comment:
in general, this class is still too complicated for what we need.  it'll work for now, but I'd prefer to whittle it way down.